app/src/main/java/com/android/unio/ui/association/AssociationProfile.kt

```diff
@@ -260,11 +267,34 @@ private fun AssociationProfileContent(
     eventViewModel: EventViewModel,
     associationViewModel: AssociationViewModel
 ) {
-  val associationState by associationViewModel.selectedAssociation.collectAsState()
-  val association = associationState!!
-  val members by association.members.list.collectAsState()
+  val association by associationViewModel.selectedAssociation.collectAsState()
   val user by userViewModel.user.collectAsState()
 
+  if (association == null || user == null) {
+    Log.e("AssociationProfileContent", "Association or user not found.")
+    return
+  }
+
+  val members by association!!.members.list.collectAsState()
+
+  var isFollowed by remember {
+    mutableStateOf(user!!.followedAssociations.contains(association!!.uid))
+  }
+  var enableButton by remember { mutableStateOf(true) }
+  val onFollow = {
+    enableButton = false
+    associationViewModel.updateFollow(association!!, user!!, isFollowed) {
+      userViewModel.refreshUser()
+      enableButton = true
+    }
+    isFollowed = !isFollowed
+  }
+
+  val onMemberClick = { member: User ->
+    userViewModel.setSomeoneElseUser(member)
+    navigationAction.navigateTo(Screen.SOMEONE_ELSE_PROFILE)
+  }
+
```

Comment:
I didn't realize most of the `UserViewModel` could be simply cleaned up to this, good catch.